Lib/test/test_capi/test_opt.py

```diff
@@ -2362,6 +2364,23 @@ def testfunc(n):
         self.assertNotIn("_GUARD_TOS_INT", uops)
         self.assertNotIn("_GUARD_NOS_INT", uops)
 
+    def test_store_fast_pop_top_specialize_unicode(self):
+        def random_str(n):
+            return ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(n))
+        def testfunc(n):
+            y = random_str(32)
+            for _ in range(n):
+                x = y + y # _POP_TOP
+                x = None # _POP_TOP_NOP
+
+        testfunc(TIER2_THRESHOLD)
+
+        ex = get_first_executor(testfunc)
+        self.assertIsNotNone(ex)
+        uops = get_opnames(ex)
+
+        self.assertIn("_POP_TOP_NOP", uops)
+
```

Comment:
Sorry I thought of a better test: you can use this one but change for float, and assert that t the `_POP_TOP_UNICODE` uop is inside https://github.com/python/cpython/blob/46b423fcd354bd229d6e430860946e1fe166bb18/Lib/test/test_capi/test_opt.py#L2310